src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -6559,18 +6561,30 @@ private bool InEnumMemberInitializer()
                 WarnOnAccessOfOffDefault(node, receiver, diagnostics);
             }
 
-            CheckRuntimeSupportForSymbolAccess(node, receiver, propertySymbol, diagnostics);
-
             return new BoundPropertyAccess(node, receiver, propertySymbol, lookupResult, propertySymbol.Type.TypeSymbol, hasErrors: (hasErrors || hasError));
         }
 
         private void CheckRuntimeSupportForSymbolAccess(SyntaxNode node, BoundExpression receiverOpt, Symbol symbol, DiagnosticBag diagnostics)
         {
-            if (!symbol.IsStatic && symbol.ContainingType.IsInterface &&
-                (!symbol.IsImplementableInterfaceMember() || (receiverOpt as BoundBaseReference)?.ExplicitBaseReferenceOpt?.Type.IsInterfaceType() == true) &&
-                !Compilation.Assembly.RuntimeSupportsDefaultInterfaceImplementation)
+            if (symbol.ContainingType?.IsInterface == true && !Compilation.Assembly.RuntimeSupportsDefaultInterfaceImplementation && Compilation.SourceModule != symbol.ContainingModule)
```

Comment:
>Compilation.SourceModule != symbol.ContainingModule [](start = 133, length = 51)

Can you please explain the reason for this condition?

Reply:
> Can you please explain the reason for this condition?

If the member is defined in the same module, then we either already complained that it is not supported at the declaration point, or it is fine to use it.  

---
In reply to: [264408195](https://github.com/dotnet/roslyn/pull/34012#discussion_r264408195) [](ancestors = 264408195)